Add some Nix-related questions to the survey

thufschmitt opened the pull request:
The three commits are independent and cover some important information for the Nix maintenance

thufschmitt committed:
Tell how people can get their version of Nix

thufschmitt committed:
Get information about how people installed Nix

GuillaumeDesforges commented on 2024/survey.yaml:
```diff
@@ -198,6 +198,12 @@ questions:
+    choices:
+      - Stability
+      - Performance
+      - New features
```
What do you mean by "new features"?

thufschmitt replied:
Getting new things added to Nix.

The goal of the question (if that makes it clearer) is for Nix maintainers to get an idea of what to prioritize between fixing bugs/increasing the reliability of the code base, making Nix faster, and bringing in the bazillion of exciting potential new developments that allow Nix to do more stuff.

GuillaumeDesforges replied:
> bringing in the bazillion of exciting potential new developments that allow Nix to do more stuff

For example?

thufschmitt replied:
I can give some concrete big examples (lazy trees, dynamic derivations, Windows support, new version of the Nix language with less corner cases, etc..), but I don't think they would be relevant because the bulk of this is really the many small improvements from https://github.com/NixOS/nix/labels/feature which don't follow any particular axis

thufschmitt replied:
To give a small but very concrete example of the problem, what I'd like to see is people's sentiment on https://github.com/NixOS/nix/pull/10203#issuecomment-1995908007

GuillaumeDesforges replied:
I think it's important to make each proposed choice more understandable. As it is the question and choices are too vague to lead to an actionnable analysis.

GuillaumeDesforges commented on 2024/survey.yaml:
```diff
@@ -253,11 +259,16 @@ questions:
       - Zig
-  - prompt: Do you use NixOS?
-    type: single
+  # Both query information about NixOS usage and the way people install Nix
```
```suggestion
```

as we delete the "Do you use NixOS?"

thufschmitt replied:
I assumed deleting the “Do you use NixOS” wouldn't loose any information since NixOS users would check the `NixOS` box in “How did you install Nix?”

Does the below make it clearer?

```suggestion
  # Query information about the installation method.
  # Also gives insight on the proportion of NixOS users since these would be the one who checked the `NixOS` box
```

GuillaumeDesforges committed:
Update 2024/survey.yaml

Co-authored-by: Théophane Hufschmitt <7226587+thufschmitt@users.noreply.github.com>

GuillaumeDesforges commented:
I'm ok merging all but [2322497](https://github.com/GuillaumeDesforges/nix-survey/pull/5/commits/2322497397163544514ec904c143669835c9950d)

thufschmitt commented:
> I'm ok merging all but [2322497](https://github.com/GuillaumeDesforges/nix-survey/pull/5/commits/2322497397163544514ec904c143669835c9950d)

I've dropped it for now. Let's discuss it separately later

GuillaumeDesforges commented:
LGTM, let's merge once conflicts are resolved :slightly_smiling_face:

thufschmitt committed:
Merge branch 'master' into patch-1

thufschmitt commented on 2024/survey.yaml:
```diff
@@ -249,9 +260,10 @@ questions:
   - prompt: Do you currently consider Nix as part of your usual set of tools?
     type: single
-    choices:
+    choice:
```
```suggestion
    choices:
```

thufschmitt commented on 2024/survey.yaml:
```diff
@@ -249,9 +260,10 @@ questions:
+    choice:
       - Yes
       - No
+
```
```suggestion
```

thufschmitt committed:
Merge resolution typos

thufschmitt commented:
>  let's merge once conflicts are resolved 

Wut? I fixed them already this morning :thinking: 

At any rate, I fixed then again, should be good now

GuillaumeDesforges commented:
Thanks a lot for your contribution :heart:
